include/nighthawk/common/factories.h

```diff
@@ -33,12 +33,21 @@ class StatisticFactory {
   virtual StatisticPtr create() const PURE;
 };
 
+class RequestSourceConstructorInterface {
+public:
+  virtual ~RequestSourceConstructorInterface() = default;
+  virtual RequestSourcePtr
+  createStaticRequestSource(Envoy::Http::RequestHeaderMapPtr&&,
+                            const uint64_t max_yields = UINT64_MAX) const PURE;
+  virtual RequestSourcePtr createRemoteRequestSource(Envoy::Http::RequestHeaderMapPtr&& base_header,
+                                                     uint32_t header_buffer_length) const PURE;
+};
+
 class RequestSourceFactory {
 public:
   virtual ~RequestSourceFactory() = default;
-  virtual RequestSourcePtr create(const Envoy::Upstream::ClusterManagerPtr& cluster_manager,
-                                  Envoy::Event::Dispatcher& dispatcher, Envoy::Stats::Scope& scope,
-                                  absl::string_view service_cluster_name) const PURE;
+  virtual RequestSourcePtr
+  create(const RequestSourceConstructorInterface& request_source_constructor) const PURE;
```

Comment:
Do we want to replace old `create()` method or add new method in addition to the old  method?

Reply:
I want to replace the old create() method. This refactor is intended to separate out the logic for how to make a requestSource into the requestSourceConstructor.